homeassistant/components/homematicip_cloud/climate.py

```diff
@@ -223,6 +223,8 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             await self._device.set_boost(False)
         if preset_mode == PRESET_BOOST:
             await self._device.set_boost()
+        if preset_mode == PRESET_ECO:
+            await self._device.set_control_mode(HMIP_ECO_CM)
```

Comment:
It would be great to test setting `PRESET_ECO` at `test_hmip_heating_group_heat`:
https://github.com/home-assistant/core/blob/dev/tests/components/homematicip_cloud/test_climate.py#L39

Something like:
```
    await hass.services.async_call(
        "climate",
        "set_preset_mode",
        {"entity_id": entity_id, "preset_mode": PRESET_ECO},
        blocking=True,
    )
    assert len(hmip_device.mock_calls) == service_call_counter + X
    assert hmip_device.mock_calls[-1][0] == "set_control_mode"
    assert hmip_device.mock_calls[-1][1] == ("HMIP_ECO_CM",)
    await async_manipulate_test_data(hass, hmip_device, "controlMode", "ECO")
    ha_state = hass.states.get(entity_id)
    assert ha_state.attributes[ATTR_PRESET_MODE] == PRESET_ECO
```